src/key/ed25519.ts

```diff
@@ -0,0 +1,39 @@
+import * as _ from "lodash";
+import nacl = require("tweetnacl");
+import { toArray, toHex } from "../utility";
+
+export type Ed25519Signature = string;
+
+/**
+ * Gets EdDSA(Ed25519) signature for message from private key.
+ * @param message 32 byte hexadecimal string
+ * @param priv 64 byte hexadecimal string of private key
+ * @returns 64 byte hexadecimal string of Ed25519 signature
+ */
+export const signEd25519 = (
+    message: string,
+    priv: string
+): Ed25519Signature => {
+    return toHex(
+        Buffer.from(nacl.sign.detached(toArray(message), toArray(priv)))
+    );
+};
+
+/**
+ * Checks if the signature from signEd25519 is correct.
```

Comment:
I think "valid" is correct rather than "correct" here. Let's modify the comments through this chance.